### PR TITLE
Vignette translating between base R and purrr

### LINKED ISSUE #726: Vignette translating to/from base R

### PR DESCRIPTION
**Pull request by lambdamoses**
Fixes #726

### CODE REVIEW

**Comment by jennybc** [comment]
Ideally you would update this PR to reflect that general roxygen2 update has happened in `master`, since you created this branch. So that the diff is more specific to your proposed vignette. Feel free to ask me or any helper if you're uncertain how to do that.

File: vignettes/base_R.Rmd [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  fig.align = "center"
+)
+```
+
+```{r setup}
+library(purrr)
+```
+
+# Introduction
+Many functions in R are vectorized, such as `as.character` and `exp`, efficiently applying the operation of interest to each element of a vector without the need to write a loop. However, some functions do not have vectorized methods for some types of vectors, especially lists (rather than atomic vectors), and sometimes the function to be applied is not vectorized in an argument of interest. In base R, the built-in `*apply` family of functions apply the function of interest to each element of a vector or list, returning the results as a list, vector, or matrix. This simplifies the code compared to writing a loop, as the need to pre-define an empty container for the results and filling that container is obliterated.
```
Comment (hadley) [inline vignettes/base_R.Rmd]: Our style is to put parentheses after function names, like `as.character()`.

File: vignettes/base_R.Rmd [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+## Shorthands
+When an anonymous function is required in `*apply` or `map` functions, `purrr` offers shorthands to make the anonymous function more readable and easier to write. However, using shorthands is not recommended inside functions and packages. Here `l` denotes a list of arguments, and `f` denotes some expression involving arguments of the anonymous function.
+
+Number of inputs | base R | `purrr`
+:---:|--------|---------
+1 | `function(x) f(x)` | `~ f(.x)`
+2 | `function(x, y) f(x, y)` | `~ f(.x, .y)`
+>2 | `function(x, y, z, ...) f(x, y, z, ...)` or `function(l) f(l[[1]], l[[2]], l[[3]], ...)` or `function(l) do.call(f, args = l)` | `~ f(..1, ..2, ..3, ...)`
+
```
Comment (hadley) [inline vignettes/base_R.Rmd]: Maybe mention the plucking shortcuts here too?

File: vignettes/base_R.Rmd (hunk 1 above)
Comment (lambdamoses) [reply vignettes/base_R.Rmd]: OK

File: vignettes/base_R.Rmd (hunk 2 above)
Comment (lambdamoses) [reply vignettes/base_R.Rmd]: OK

**Comment by hadley** [comment]
Thanks for working on this and sorry it took so long to merge!

**Pull request merged**